### ISSUE TITLE
QUIC support for `Acceptor`

### ISSUE THREAD

**Issue by daxpedda** [rustls/rustls#1218]
Currently the `Acceptor` API can't be used with QUIC. It's missing something like [`new_quic()`](https://docs.rs/rustls/0.20.8/rustls/server/struct.ServerConnection.html#method.new_quic) among other things.

I actually wanted to work on this myself quite some time ago but got distracted with a bunch of other stuff. Maybe I will get to it again, thought I would leave that here considering #1216.

This was briefly discussed a bit more then a year ago on matrix:
https://matrix.to/#/!kZVoAlfRzfachIZeCA:ralith.com/$bu5HQXTF-eJWBqQhl0RrxlWQc0x9a7tUD-OTDSR7eDQ?via=mozilla.org&via=gitter.im&via=matrix.org

**Comment by djc** [comment]
Given that this would be new API, I think we can postpone this past the upcoming 0.21 release. But would be great if you can work on this after #1216 is merged!

**Comment by cpu** [comment]
@daxpedda We've stabilized the new API from #1216 and released v0.21.0. Are you able to work on this feature request?

**Comment by daxpedda** [comment]
If you are interested in working on this yourself please go ahead, it will take me some time to get back to this.

**Comment by ctz** [comment]
Closed via #3070 -- thank you @iadev09 !

**Comment by iadev09** [comment]
Thank you [ctz](https://github.com/ctz) for your great work — for a better world.
